gtsam/linear/NoiseModel.h

```diff
@@ -839,6 +860,16 @@ namespace gtsam {
           }
           return 0.0;
         }
+        double residual(double error) const {
+          double absError = std::abs(error);
+          if (absError <= c_) {
+            double t = csquared_ - (error*error);
+            double val = t * t * t;
+            return ((csquared_*csquared_*csquared_) - val) / (6.0 * csquared_ * csquared_);
```

Comment:
So, c^6 ?? Seems weird. Is this validated by a test? Also, should be computed in constructor then.

Reply:
I modified this to make this easier to read and correlate with the actual expression.